Islandora-Devops/isle-dc
# Configured drupal

Spins up a properly configured Drupal and Houdini.  Note that this is pointing at a local copy of https://github.com/Islandora-Devops/isle-houdini.  Let me know what I need to change once https://github.com/Islandora-Devops/isle-houdini/pull/3 has been merged.

## Testing instructions

- Get into the directory
  - `cd /path/to/isle-dc`
- Kill whatever you may have
  - `docker-compose down -v`
- Clear out the codebase if it already exists
  - `rm -rf codebase; mkdir codebase`
- Generate keys if you don't have them
  - `(cd scripts; ./generate_keys.sh)`
- Build the images.  This will template out config for houdini and deploy it.
  - `docker-compose up --build -d`
- Run some commands to set up solr and drupal
  - `docker exec -it isle_dc_proto_solr bash -c "solr create_core -c islandora"`
  - `docker exec -it isle_dc_proto_php bash -c "sh /scripts/islandora/install-islandora.sh"`
- Visit idcp.localhost for a somewhat functional Drupal site (you can ignore the bit about `continue 2`, it's harmless and we'll address that later)
- Use houdini by doing something like
  - `curl -H "Authorization: Bearer islandora" -H "Apix-Ldp-Resource: https://islandora.ca/sites/default/files/Islandora.png" idcp.localhost:8000/identify`
  - That should return the json output for the identify command on the islandora logo
